src/test/java/teammates/test/cases/webapi/GetFeedbackQuestionRecipientsActionTest.java

```diff
@@ -23,12 +52,190 @@ protected String getRequestMethod() {
     @Test
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+
+        ______TS("Invalid intend, should fail");
+        String[] invalidIntendParams = generateParameters(firstSessionInCourse1, 1, Intent.FULL_DETAIL,
+                "", "", "");
+        verifyHttpParameterFailure(invalidIntendParams);
+
+        testDifferentRecipientTypes();
+        testModeratedAndPreviewPersonToGetRecipients();
+    }
+
+    private String[] generateParameters(FeedbackSessionAttributes session, int questionNumber, Intent intent,
+                                        String regKey, String moderatedPerson, String previewPerson) {
+        FeedbackQuestionAttributes question = logic.getFeedbackQuestion(session.getFeedbackSessionName(),
+                session.getCourseId(), questionNumber);
+        return new String[] {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, question.getFeedbackQuestionId(),
+                Const.ParamsNames.INTENT, intent.toString(),
+                Const.ParamsNames.FEEDBACK_SESSION_MODERATED_PERSON, moderatedPerson,
+                Const.ParamsNames.PREVIEWAS, previewPerson,
+                Const.ParamsNames.REGKEY, regKey,
+        };
+    }
+
+    private FeedbackQuestionRecipientsData getRecipients(String[] params) {
+        verifyCanAccess(params); // Make sure not violates access control
+        GetFeedbackQuestionRecipientsAction action = getAction(params);
+        JsonResult result = getJsonResult(action);
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        return (FeedbackQuestionRecipientsData) result.getOutput();
+    }
+
+    private void verifyFeedbackQuestionRecipientsDataEquals(FeedbackQuestionRecipientsData expected,
+                                                            FeedbackQuestionRecipientsData actual) {
+        assertEquals(expected.getRecipients().size(), actual.getRecipients().size());
+        for (int i = 0; i < expected.getRecipients().size(); i++) {
+            assertEquals(expected.getRecipients().get(i).getIdentifier(),
+                    actual.getRecipients().get(i).getIdentifier());
+            assertEquals(expected.getRecipients().get(i).getName(), actual.getRecipients().get(i).getName());
+        }
+    }
+
+    private void testModeratedAndPreviewPersonToGetRecipients() {
```

Comment:
Instead of calling this method directly in testExecute, add a @Test before the method name to mark it as an individual test. See https://github.com/TEAMMATES/teammates/blob/master/docs/best-practices/testing.md
```suggestion
    @Test
    private void testModeratedAndPreviewPersonToGetRecipients() {
```

Reply:
👍 Note the naming convention also.